test/go-tests/test_utils.go

```diff
@@ -1141,3 +1145,105 @@ func resetTestPath(t *testing.T, path string) {
 	err := os.Chdir(path)
 	require.Nil(t, err)
 }
+
+func createZipFileFromDirectory(source, target string, needBaseDir bool) error {
+	zipfile, err := os.Create(target)
+	if err != nil {
+		return err
+	}
+	defer zipfile.Close()
+
+	archive := zip.NewWriter(zipfile)
+	defer archive.Close()
+
+	info, err := os.Stat(source)
+	if err != nil {
+		return err
+	}
+
+	var baseDir string
+	if info.IsDir() {
+		baseDir = filepath.Base(source)
+	}
+
+	filepath.Walk(source, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		header, err := zip.FileInfoHeader(info)
+		if err != nil {
+			return err
+		}
+
+		if baseDir != "" {
+			if needBaseDir {
+				header.Name = filepath.Join(baseDir, strings.TrimPrefix(path, source))
+			} else {
+				path := strings.TrimPrefix(path, source)
+				if len(path) > 0 && (path[0] == '/' || path[0] == '\\') {
+					path = path[1:]
+				}
+				if len(path) == 0 {
+					return nil
+				}
+				header.Name = path
+			}
+		}
+
+		if info.IsDir() {
+			header.Name += "/"
+		} else {
+			header.Method = zip.Deflate
+		}
+
+		writer, err := archive.CreateHeader(header)
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		file, err := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+		_, err = io.Copy(writer, file)
+		return err
+	})
+
+	return err
+}
+
+func checkIfSecretExists(secretName string) error {
```

Comment:
maybe lets change the signature of this to `func(secretName, namespace)`
from the caller `checkIfSecretExists("mysecret",GetKeptnNamespaceFromEnv())` seems to be still comforable enough to me...